apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -313,20 +313,25 @@ impl Plugin for Telemetry {
                 response
             })
             .instrument(move |request: &router::Request| {
-                let apollo = config.apollo.as_ref().cloned().unwrap_or_default();
+                let router_request = &request.router_request;
+                let headers = router_request.headers();
+                let default_forward = ForwardHeaders::default();
+                let (client_name, client_version, send_headers ) = config.apollo.as_ref().map(|apollo| {
+                    (
+                        headers
+                            .get(&apollo.client_name_header).and_then(|h| h.to_str().ok()).unwrap_or(""),
+                        headers
+                            .get(&apollo.client_version_header)
+                            .and_then(|h| h.to_str().ok())
+                            .unwrap_or(""),
+                        &apollo.send_headers)
+
+                }).unwrap_or_else(||("", "", &default_forward));
+
```

Comment:
is this the expected behaviour? If the apollo config is not there, not getting the client name and version from headers? We can have cases where apollo key and graph ref are present but not the config, so that's probably not what we want

Reply:
solved in [44904f1](https://github.com/apollographql/router/pull/3887/commits/44904f15c09b4a5019e3404d24e276b4175f00ef)